### ISSUE TITLE
Viewing some accounts redirects away from one's own instance to the remote instance

### ISSUE THREAD

**Issue by mlncn** [mastodon/mastodon#24531]
### Steps to reproduce the problem

1. Log into social.coop or the Mastodon instance of your choice
2. Click the profile or try to load the profile for @commscholar@federate.social in the last column (action/active column, where you see the post, thread, or profile you press in another column), that is, you are (for a moment) at the URL like https://social.coop/@commscholar@federate.social 
3. You get redirected to https://federate.social/@commscholar which makes it hard to follow or otherwise interact with this profile!

This is the expected behavior when logged out so it's not that easy to test.

I have experienced this on servers running Mastodon v4.1.2 and v4.1.0 at least.

### Expected behaviour

When i visit a profile while logged in, i always stay within my home instance

### Actual behaviour

Sometimes i am redirected to the profile on its own instance

### Detailed description

That's all i have.  Been happening for a while and hoping to see someone else note it or for it to resolve itself but it hasn't, so here goes this bug report.

### Specifications

v4.1.2 and v4.1.0
Firefox

**Comment by mlncn** [comment]
That particular profile is already working for me to visit, from both instances of Mastodon that it had not worked on when i filed this.

Did toggle the Firefox extension "uBlock origin" off for *one* of those instances but not the other, and the profile now works on both.

My best guess is it is a behavior Mastodon somehow does accidentally when the instance i'm on doesn't have an established connection of some kind to the profile's instance?  So maybe people on small instances can reproduce this?

**Comment by mlncn** [comment]
Maybe try @Articles@portside.org as most no one should have any connection to that server?

**Comment by mlncn** [comment]
One other person reports experiencing this but no one has been able to reproduce :shrug:

**Comment by konomikitten** [comment]
I've encountered this occurring on mastodon.online and it's extremely annoying.

### Specifications

Mastodon v4.1.2
Firefox 102.10.0esr

**Comment by MaddiFurr** [comment]
I too have been having this issue while using the furry.engineer mastodon instance

### Specs

Mastodon v4.1.2
Firefox 114.0.1

When I go to someone's profile on another instance, even if I search for them first, if I try to interact with a post of theirs, it will open a new tab where it will try to direct me to mastodon.social. A similar issue happens when I'm on my profile if I hit the main Edit Profile button (xpath: `/html/body/div[1]/div/div/div[2]/div[2]/div[2]/div/div/div/div/div[1]/div[2]/div[1]/div/button`) it will also redirect me to mastodon.social rather than the edit profile page of the instance I am a part of.
